tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs

```diff
@@ -470,50 +472,53 @@ public async Task HandlePlaybackRequest(string recordingId, HttpRequest incoming
 
             var entry = (await CreateEntryAsync(incomingRequest).ConfigureAwait(false)).Item1;
 
-            // Session may be removed later, but only after response has been fully written
-            var match = session.Session.Lookup(entry, session.CustomMatcher ?? Matcher, sanitizers, remove: false);
-
-            foreach (ResponseTransform transform in Transforms.Concat(session.AdditionalTransforms))
+            lock (session.Session.Entries)
```

Comment:
Did you try adding this lock, but still writing the body bytes async?  I can see potential race conditions fixed by locking all entries until all bytes have been written, but I don't think sync vs async WriteBodyBytes should matter.

Reply:
As far as I'm aware you can't write await within a `lock`?

Am I crazy? I'll try it!

Reply:
<img width="692" alt="image" src="https://github.com/user-attachments/assets/db7ac6ed-5529-49d5-b68a-5ab80b1b6718">

Reply:
It can't be done using the `lock` statement itself, but conceptually it's possible to use locking semantics with async:

https://stackoverflow.com/a/18257065

Reply:
Understood! I've definitely used a SempahoreSlim before, I'll apply that and see if the experience is reproed. if so, I'll keep the async around.